traitlets/config/loader.py

```diff
@@ -598,6 +604,69 @@ def __exit__(self, exc_type, exc_value, traceback):
             f.write(json_config)
 
 
+class TOMLFileConfigLoader(FileConfigLoader):
+    """A TOML file loader for config
+
+    Can also act as a context manager that rewrite the configuration file to disk on exit.
+
+    Example::
+
+        with TOMLFileConfigLoader('myapp.toml','/home/jupyter/configurations/') as c:
+            c.MyNewConfigurable.new_value = 'Updated'
+
+    """
+
+    def __init__(self, filename, **kw):
+        """Wrapper for checking (optional) toml module import"""
+        if not HAS_TOML:
+            raise ConfigLoaderError('toml module is not found. In order to use toml configuration'
+                                    'files, please either install traitlets with corresponding option'
+                                    ' (pip install "traitlets[with-toml]") or simply add it with'
+                                    '"pip install toml"')
+        super(TOMLFileConfigLoader, self).__init__(filename, **kw)
+
+    def load_config(self):
+        """Load the config from a file and return it as a Config object."""
+        self.clear()
+        try:
+            self._find_file()
+        except IOError as e:
+            raise ConfigFileNotFound(str(e))
+        dct = self._read_file_as_dict()
+        self.config = self._convert_to_config(dct)
+        return self.config
+
+    def _read_file_as_dict(self):
+        with open(self.full_filename) as f:
+            return toml.load(f)
+
+    def _convert_to_config(self, dictionary):
+        if 'version' in dictionary:
```

Comment:
i don't recall having seen `version` on any of the other sources, and don't see why we'd want to add that constraint here unless it was added to all of them, which would be a major version bump